models/keyword.go

```diff
@@ -22,14 +54,29 @@ func GetKeywordsBy(condition map[string]interface{}) ([]Keyword, error) {
 	return keywords, nil
 }
 
-func SaveKeywords(keywords []Keyword) ([]Keyword, error) {
-	// Insert bulk data
-	result := db.GetDB().Create(&keywords)
+func SaveKeyword(keyword Keyword, tx ...*gorm.DB) (Keyword, error) {
+	var cnx *gorm.DB
+	if tx != nil {
+		cnx = tx[0]
```

Comment:
That's the strangest part of the transaction implementation as I am not sure why we need to select the first item in the tx slide argument 🤔

Reply:
Function that receive parameter like this `(tx ...*gorm.DB) called Variadic function and that allow you to pass 0 or more parameter to the function.
Here is the simple example
https://gobyexample.com/variadic-functions

Reply:
I understood the argument type, but why I am not sure why we always pick the first item in the tx param 🙈

Reply:
In the first place, I thought I want transaction to be optional that's why I use variadic param but that's not make sense so Fixed in 1807813c580b9a0df5262c564930e63b3c9b8bd2